client/my-sites/checkout/checkout-thank-you/pending/index.tsx

```diff
@@ -354,12 +355,14 @@ function triggerPostRedirectNotices( {
 		return;
 	}
 
-	reduxDispatch(
-		successNotice( translate( 'Your purchase has been completed!' ), {
-			id: 'checkout-thank-you-success',
-			displayOnNextPage: true,
-		} )
-	);
+	if ( ! isRefactoredForThankYouV2 ) {
+		reduxDispatch(
+			successNotice( translate( 'Your purchase has been completed!' ), {
+				id: 'checkout-thank-you-success',
+				displayOnNextPage: true,
+			} )
+		);
+	}
```

Comment:
Here is another approach for your consideration:

```javascript
if ( isRefactoredForThankYouV2 ) {
  // Disables notice as it is redundant with the content of the new Thank You page
  return;
}

reduxDispatch(
  successNotice( translate( 'Your purchase has been completed!' ), {
    id: 'checkout-thank-you-success',
    displayOnNextPage: true,
  } )
);
```

Reply:
I would probably get rid of the `isRefactoredForThankYouV2` check. We should be good to remove this notice for both legacy and redesign Congrats Page. And in reality, I think the majority (if not all) of purchases land on the redesigned page. As discussed in Slack, I'm not even sure if the "delayed transfer" flow is valid at this point.

* When users purchase a paid plan + domain transfer, they are taken to the "What is your goal?" flow
* Users cannot purchase domain transfer when they select a free plan.

See p1712689984399789/1712687032.878589-slack-C06ELHR6L9J

Reply:
I've added a new change that just removes the notification all together. I'll wait for CI and a final approval from you @wongasy!